src/scraper/scraper.ts

```
@@ -0,0 +1,98 @@
+export function hasTimeLabel(labels: string[]): boolean {
+  return labels.some((label) => TIME_LABEL_PATTERNS.some((p) => p.test(label)));
+}
+
+export function extractTimeLabel(labels: string[]): string | null {
+  return labels.find((label) => TIME_LABEL_PATTERNS.some((p) => p.test(label))) ?? null;
+}
+
+/**
+ * Fetch issues from a GitHub repository that have time-related labels.
+ */
+export async function scrapeIssues(
```

ESLint

[error] Refactor this function to reduce its Cognitive Complexity from 26 to the 15 allowed.

(sonarjs/cognitive-complexity)